sources/packages/backend/apps/api/src/services/application-exception/application-exception.service.ts

```diff
@@ -63,12 +65,19 @@ export class ApplicationExceptionService extends RecordDataModelService<Applicat
       .leftJoin("exception.assessedBy", "assessedBy")
       .where("exception.id = :exceptionId", { exceptionId });
 
+    if (studentId || applicationId) {
+      exception.innerJoin("exception.application", "application");
+    }
+
     if (studentId) {
       exception
-        .innerJoin("exception.application", "application")
         .innerJoin("application.student", "student")
         .andWhere("student.id = :studentId", { studentId });
     }
+
+    if (applicationId) {
+      exception.andWhere("application.id = :applicationId", { applicationId });
```

Comment:
I'm confused here. I was in an assumption that, with decorators in place we will not be again doing the same applicationId condition in query in the services.

Reply:
not sure, if I got your question, the decorator will check if the student has access to the student id and application. each service will check if the entity is related to the application and student here, `exception` is related to the application and student.

Reply:
I understood the point later. Thanks for explaining.